src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -112,7 +112,7 @@ public override void OnStartActivity(Activity activity, object payload)
 
             if (activity.IsAllDataRequested)
             {
-                activity.SetCustomProperty(RequestCustomPropertyName, request);
+                this.options.Enrich?.Invoke(activity, "OnStartActivity", request);
```

Comment:
this can be user code and can potentially throw. We need to try.catch.log

Reply:
Should I add for the entire method as a protection or just for this invoke?

Reply:
At the minimum - do it when we call Enrich, as it can be user code, so anything could happen.

We can review if we need to try.catch the whole method. (We may need it, as the potential impact of an unhandled exception is very bad) - But this can be separate, to reduce scope of this PR.

Reply:
done